packages/block-library/src/cover/edit/index.js

```diff
@@ -168,7 +170,16 @@ function CoverEdit( {
 			} );
 		} )();
 		// Update the block only when the featured image changes.
-	}, [ mediaUrl ] );
+	}, [
+		useFeaturedImage,
+		mediaUrl,
+		overlayColor.color,
+		isUserOverlayColor,
+		dimRatio,
+		__unstableMarkNextChangeAsNotPersistent,
+		setOverlayColor,
+		setAttributes,
```

Comment:
I think the change here is probably unintended right? It feels a bit risky

Reply:
+1. The effect deps array also has a note - "Update the block only when the featured image changes."